FakebookNotifications/FakebookNotifications.Testing/IntegrationTests/NotificationControllerTest.cs

```diff
@@ -0,0 +1,188 @@
+            };
+        }
+
+        private readonly Domain.Models.User testUser1 = new()
+        {
+            Id = "01",
+            Connections = new List<string>{
+                "00", "02", "03"
+            },
+            Email = "test@test.com"
+        };
+
+        private readonly List<string> groupIds = new()
+        {
+            "test@test.com",
+            "group1",
+            "group2",
+            "group3"
+        };
+
+        private readonly List<string> clientIds = new() { "00", "01", "02", "03", "04", "05" };
+
+        private readonly Mock<IHubCallerClients> mockClients = new();
+        private readonly Mock<IGroupManager> mockGroups = new();
+        private readonly Mock<IClientProxy> mockClientProxy = new();
+        private readonly Mock<HubCallerContext> mockContext = new();
+        private readonly NotificationHub mockHub;
+
+        private readonly UserRepo _userRepo;
+        private readonly NotificationsRepo _noteRepo;
+        private readonly NotificationsDatabaseSettings settings;
+        private readonly NullLogger<NotificationsContext> _logger;
+
+        public NotificationControllerTest()
+        {
+            //mocking signalr elements for tests
+            mockClients.Setup(client => client.All).Returns(mockClientProxy.Object);
+            mockClients.Setup(client => client.Group(groupIds[0])).Returns(mockClientProxy.Object);
+            mockClients.Setup(client => client.Caller).Returns(mockClientProxy.Object);
+            mockClients.Setup(client => client.OthersInGroup(It.IsIn<string>(groupIds))).Returns(mockClientProxy.Object);
+            mockGroups.Setup(group => group.AddToGroupAsync(It.IsIn<string>(clientIds), It.IsIn<string>(groupIds), new CancellationToken())).Returns(Task.FromResult(true));
+            mockGroups.Setup(group => group.RemoveFromGroupAsync(It.IsIn<string>(clientIds), It.IsIn<string>(groupIds), new CancellationToken())).Returns(Task.FromResult(true));
+            mockContext.Setup(context => context.ConnectionId).Returns("01");
+            
+
+            _logger = new NullLogger<NotificationsContext>();
+            Mock<IOptions<NotificationsDatabaseSettings>> mockSettings = new();
+            settings = new NotificationsDatabaseSettings
+            {
+                ConnectionString = "mongodb+srv://ryan:1234@fakebook.r8oce.mongodb.net/Notifications?retryWrites=true&w=majority",
+                //ConnectionString = "mongodb://localhost:27017",
+                DatabaseName = "Notifications",
+                UserCollection = "User",
+                NotificationsCollection = "Notifications"
+            };
+            mockSettings.Setup(s => s.Value).Returns(settings);
+            NotificationsContext mockDbContext = new(mockSettings.Object, _logger);
+
+            // mocking Mongo db
+            NotificationsRepo noteRepo = new(mockDbContext);
+            UserRepo userRepo = new(mockDbContext, noteRepo);
```

Comment:
the integration test includes these objects too? yeah, i guess you could write an integration test that spans all those classes. but it's certainly not mocking.